### PR TITLE
feat: added keycloak operator

### PR DESCRIPTION
**Pull request by ElderMatt**
This PR removed the keycloak job and replaces it with an operator.

### CODE REVIEW

**Review by ferruhcihan — APPROVED** [review]
LGTM

File: charts/apl-keycloak-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      serviceAccountName: {{ include "apl-keycloak-operator.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if and .Values.configs (hasKey .Values.configs "tls") (hasKey .Values.configs.tls "certificates") }}
```
Comment (j-zimnowoda) [inline charts/apl-keycloak-operator/templates/deployment.yaml:34]: From the values.yaml file:
```
configs:
  tls: 
    certificates: {}
```

the path `configs.tls.certificates` seems to always exist. Wouldn't be better to check if the `certificates` is not an empty string?

File: charts/apl-keycloak-operator/templates/apl-keycloak-operator-tls-certs-cm.yaml
Comment (j-zimnowoda) [inline charts/apl-keycloak-operator/templates/apl-keycloak-operator-tls-certs-cm.yaml:1]: It is customary to name in just configmap.yaml if you have just a single cm

File: charts/apl-keycloak-operator/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list"]
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["get", "watch", "list"]
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["get", "watch", "list"]
+- apiGroups: [""]
```
Comment (j-zimnowoda) [inline charts/apl-keycloak-operator/templates/rbac.yaml:23]: Do the operator need to read secrets from all namespaces?

File: charts/apl-keycloak-operator/templates/tests/test-connection.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (j-zimnowoda) [inline charts/apl-keycloak-operator/templates/tests/test-connection.yaml:1]: operator is not a server thus connection does not need to be tested

File: charts/apl-keycloak-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+image:
+  repository: otomi/tasks
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: main
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+service:
```
Comment (j-zimnowoda) [inline charts/apl-keycloak-operator/values.yaml:16]: there is no service for the operator

File: values/apl-keycloak-operator/apl-keycloak-operator-raw.gotmpl [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+{{- $o := $v.apps | get "apl-keycloak-operator" }}
+{{- $version := $v.versions.tasks }}
+{{- $isSemver := regexMatch "^[0-9.]+" $version }}
+{{- $k := $c | get "keycloak" }}
+{{- $doms := tpl (readFile "../../helmfile.d/snippets/domains.gotmpl") $v | fromYaml }}
+{{- $joinTpl := readFile "../../helmfile.d/utils/joinListWithSep.gotmpl" }}
+
+resources:
+{{- if $v._derived.untrustedCA }}
+  - apiVersion: v1
```
Comment (j-zimnowoda) [inline values/apl-keycloak-operator/apl-keycloak-operator-raw.gotmpl:24]: The `custom-ca` does not seem to be used

File: charts/apl-keycloak-operator/templates/apl-keycloak-operator-tls-certs-cm.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
```
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply charts/apl-keycloak-operator/templates/apl-keycloak-operator-tls-certs-cm.yaml:2]: TLS certificates are usually stored in secrets not configmaps

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Hi there few comments about the Helm chart

**Review by j-zimnowoda — APPROVED** [review]

**Pull request merged**